src/Compilers/CSharp/Portable/Symbols/Synthesized/RefKindVector.cs

```diff
@@ -69,5 +78,65 @@ public override int GetHashCode()
         {
             return _bits.GetHashCode();
         }
+
+        public string ToRefKindString()
+        {
+            var pooledBuilder = PooledStringBuilder.GetInstance();
+            var builder = pooledBuilder.Builder;
+
+            builder.Append("{");
+
+            int i = 0;
+            foreach (int byRefIndex in this.Words())
+            {
+                if (i > 0)
+                {
+                    builder.Append(",");
+                }
+
+                builder.AppendFormat("{0:x8}", byRefIndex);
+                i++;
+            }
+
+            builder.Append("}");
+            Debug.Assert(i > 0);
+
+            return pooledBuilder.ToStringAndFree();
+        }
+
+        public static bool TryParse(string refKindString, int capacity, out RefKindVector result)
+        {
+            ulong? firstWord = null;
+            List<ulong>? otherWords = null;
+            foreach (var word in refKindString.Split(','))
+            {
+                ulong value;
+                try
+                {
+                    value = Convert.ToUInt64(word, 16);
+                }
+                catch (Exception ex) when (FatalError.ReportAndPropagate(ex))
```

Comment:
>when (FatalError.ReportAndPropagate(ex)) [](http://example.com/codeflow?start=37&length=40)

I don't think we need to log NFW for this. It's just bad input to the compiler, not a bug in the compiler.